grr/server/grr_response_server/output_plugins/elasticsearch_plugin_test.py

```diff
@@ -0,0 +1,213 @@
+#!/usr/bin/env python
+# Lint as: python3
+# -*- encoding: utf-8 -*-
+"""Tests for Elasticsearch output plugin."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import unicode_literals
+
+from unittest import mock
+
+from absl import app
+import requests
+
+from grr_response_core.lib import rdfvalue
+from grr_response_core.lib.rdfvalues import client as rdf_client
+from grr_response_core.lib.rdfvalues import client_fs as rdf_client_fs
+from grr_response_core.lib.rdfvalues import flows as rdf_flows
+from grr_response_core.lib.rdfvalues import paths as rdf_paths
+from grr_response_core.lib.util.compat import json
+from grr_response_server import data_store
+from grr_response_server.output_plugins import elasticsearch_plugin
+from grr_response_server.rdfvalues import flow_objects as rdf_flow_objects
+from grr.test_lib import flow_test_lib
+from grr.test_lib import test_lib
+
+KWARGS = 1
+
+
+class ElasticsearchOutputPluginTest(flow_test_lib.FlowTestsBaseclass):
+  """Tests Elasticsearch hunt output plugin."""
+
+  def setUp(self):
+    super().setUp()
+
+    self.client_id = self.SetupClient(0)
+    self.flow_id = '12345678'
+    data_store.REL_DB.WriteFlowObject(
+        rdf_flow_objects.Flow(
+            flow_id=self.flow_id,
+            client_id=self.client_id,
+            flow_class_name='ClientFileFinder',
+            create_time=rdfvalue.RDFDatetime.Now(),
+        ))
+
+  def _CallPlugin(self, plugin_args=None, responses=None, patcher=None):
+    source_id = rdf_client.ClientURN(
+        self.client_id).Add('Results').RelativeName('aff4:/')
+
+    messages = []
+    for response in responses:
+      messages.append(
+          rdf_flows.GrrMessage(
+              source=self.client_id,
+              session_id='{}/{}'.format(self.client_id, self.flow_id),
+              payload=response))
+
+    plugin_cls = elasticsearch_plugin.ElasticsearchOutputPlugin
+    plugin, plugin_state = plugin_cls.CreatePluginAndDefaultState(
+        source_urn=source_id, args=plugin_args)
+
+    if patcher is None:
+      patcher = mock.patch.object(requests, 'post')
+
+    with patcher as patched:
+      plugin.ProcessResponses(plugin_state, messages)
+      plugin.Flush(plugin_state)
+      plugin.UpdateState(plugin_state)
+
+    return patched
+
+  def _ParseEvents(self, patched):
+    request = patched.call_args[KWARGS]['data']
+
+    # Elasticsearch bulk requests are line-deliminated pairs, where the first
+    # line is the index command and the second is the actual document to index
+    split_request = request.split('\n')
+    print(split_request)
+    update_pairs = [(split_request[i], split_request[i + 1])
+                    for i in range(0, len(split_request), 2)]
+    parsed_pairs = [(json.Parse(i[0]), json.Parse(i[1])) for i in update_pairs]
```

Comment:
Instead of deconstructing the tuple, parsing the elements, and reconstructing a tuple, could you call `json.Parse` on the whole list in `split_requests`? This might simplify the code a little, e.g.:

```py
split_request = [json.Parse(line) for line in request.split("\n")]
```